python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -1247,29 +1254,20 @@ def diff(self, periods=1, axis=0):
         if not axis == 0:
             raise NotImplementedError("Only axis=0 is supported.")
 
-        # grouped values
-        value_columns = self.grouping.values
-        _, (data, index), _ = self._groupby.groups(
-            cudf.core.frame.Frame(value_columns._data)
+        values = self.obj.__class__._from_data(
+            self.grouping.values._data, self.obj.index
         )
-        grouped = self.obj.__class__._from_data(data, index)
-        grouped = self._mimic_pandas_order(grouped)._copy_type_metadata(
-            value_columns
-        )
-
-        result = grouped - self.shift(periods=periods)
-        return result._copy_type_metadata(value_columns)
+        return values - self.shift(periods=periods)
```

Comment:
Note: the old code here retrieves the grouped value and then calls `_mimic_pandas_order` to reconstruct original order - this is an unnecessary kernel launch. We can just get the value columns via `self.grouping.values`. Similar pattern was also found in `fillna`.